packages/react/src/grouped-fields/GroupedFields.tsx

```diff
@@ -0,0 +1,127 @@
+import { ReactNode } from 'react';
+import { visuallyHiddenStyles } from '../a11y';
+import { Box } from '../box';
+import { mapSpacing, useId } from '../core';
+import { FieldContainer, FieldHint, FieldLabel, FieldMessage } from '../field';
+import { Flex } from '../flex';
+import { Stack } from '../stack';
+
+type FieldA11yProps = {
+	'aria-describedby': string | null;
+	'aria-invalid': boolean;
+};
+
+export type GroupedFieldsProps = {
+	/** The two form inputs. */
+	children: (props: {
+		field1Props: FieldA11yProps;
+		field2Props: FieldA11yProps;
+	}) => ReactNode;
+	/** If true, the invalid state will be rendered for field 1. */
+	field1Invalid?: boolean;
+	/** If true, the invalid state will be rendered for field 2. */
+	field2Invalid?: boolean;
```

Comment:
Thanks for the input, but I'll try to state our reasoning for the decisions. We currently only ever plan to support 2 fields. This is component is purely for the known use cases like weight & unit, and date & time.

Here's our WIP design docs: [Grouping fields](https://www.figma.com/file/VjfFBpZsSddcm7pFGcc9CU/Component-development-and-testing?type=design&node-id=10297-70120&mode=design&t=NrP4o7J96vU9WeR3-0) and [Date and time](https://www.figma.com/file/VjfFBpZsSddcm7pFGcc9CU/Component-development-and-testing?type=design&node-id=10284-68366&mode=design&t=nLiUtbCGaQlZwXoI-0)

Opening it up to more fields will no doubt open the floodgates of teams trying to recreate existing UIs where they have 3 or 4 fields on the same line… not because they're particularly related, but because they want to save space. We do not want to encourage this. By restricting it to 2, we should prevent a lot of potential misuse.

We also want to have simple, as-consistent APIs as possible throughout our DS. All of our fields currently accept an `invalid` boolean prop, so it makes sense for this extension to accept 2 boolean props, which meant I had the choice of `invalidFieldX` and `fieldXInvalid`. I opted for the latter since it reads better, even though the former may be slightly more consistent with the original props.

A single prop is either true or false whereas an `invalidFields` prop is not. If one is true and the other is false like in your example, then technically it's incorrect because this is not a list of invalid fields. You could opt to only pass the invalid ones, but then a consumer has to do extra work. You could workaround that with naming like `fieldValidity`, but then that fails the consistency test.

Also, there is also no `id` to speak of, so we would have to restrict the keys used anyway, which defeats the purpose of your suggestion. And because an object is not a primitive, it will be slightly less performant because it won't be able to shallow compare, forcing us to memoise, which is more effort for the consumer and more intensive for the user.

Reply:
ok I better understand your reasoning now.

It seems feasible that  `date`, `time`, and `time-zone` fields being all displayed at once would be a valid use case for requiring 3 fields.

Reply:
As discussed, if we're asking for timezones, we'll more than likely ask for it first, so that when a user then reaches the calendar, all dates are correct.